session.go

```diff
@@ -129,6 +130,7 @@ type session struct {
 	// goroutines sync
 	grNum uatomic.Int32
 	lock  sync.RWMutex
+	wlock sync.Mutex
```

Comment:
u should use this wlock before every write action/funcs.

Reply:
addressed